common/src/main/java/net/draycia/carbon/common/command/commands/NicknameCommand.java

```diff
@@ -125,8 +126,19 @@ private void resetNickname(final Commander sender, final CarbonPlayer target) {
     }
 
     private void applyNickname(final Commander sender, final CarbonPlayer target, final String nick) {
+
         // Lazy since the sender might not have permission to set the nickname
         final Supplier<Component> parsedNick = Suppliers.memoize(() -> parseNickname(sender, nick));
+
+        // If the nickname is caught in the character limit, return without setting a nickname.
+        final int nickNameLength = PlainTextComponentSerializer.plainText().serialize(parsedNick.get()).length();
```

Comment:
I think we want to check #codePoints instead to account for multi-byte characters.